Makefile

```diff
@@ -10,16 +10,28 @@ run_migrations: virtualenv
 	${VIRTUALENV_ROOT}/bin/python application.py db upgrade
 
 virtualenv:
-	[ -z $$VIRTUAL_ENV ] && [ ! -d venv ] && virtualenv venv || true
+	[ -z $$VIRTUAL_ENV ] && [ ! -d venv ] && virtualenv -p python3 venv || true
 
 bootstrap: virtualenv
 	./scripts/bootstrap.sh
 
 requirements: virtualenv requirements.txt
 	${VIRTUALENV_ROOT}/bin/pip install -r requirements.txt
 
-requirements_for_test: virtualenv requirements_for_test.txt
-	${VIRTUALENV_ROOT}/bin/pip install -r requirements_for_test.txt
+requirements-dev: virtualenv requirements-dev.txt
+	${VIRTUALENV_ROOT}/bin/pip install -r requirements-dev.txt
+
+freeze-requirements:
+	rm -rf venv-freeze
+	virtualenv -p python3 venv-freeze
+	$$(pwd)/venv-freeze/bin/pip install -r requirements-app.txt
+	echo '# This file is autogenerated. Run `make freeze-requirements`' > requirements.txt
+	echo '# to update it with any changes made in requirements-app.txt' >> requirements.txt
+	echo '' >> requirements.txt
+	cat requirements-app.txt >> requirements.txt
+	echo '' >> requirements.txt
+	$$(pwd)/venv-freeze/bin/pip freeze -r requirements-app.txt | sed -n '/The following requirements were added by pip freeze/,$$p' >> requirements.txt
```

Comment:
We pass a requirements file to pip freeze in `pip freeze -r requirements-app.txt`, which makes it keep the packages listed in the file in place and append everything else. The problem is that it also loses git package urls, replacing them with package names. So instead, I copy the original file and then only add the additional packages from the pip freeze output (the sed command skips all lines until it matches the pip freeze comment about added packages).